helpdesk_mgmt_ticket_autoarchive/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,19 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import fields, models
+import datetime
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = 'helpdesk.ticket'
+    
+    def autoarchive_tickets(self):
+        
+        interval_number = self.env.user.company_id.autoarchive_tickets
+        date_archive = datetime.datetime.now() - datetime.timedelta(days=interval_number)
+        closed = self.env["helpdesk.ticket"].search(["&",("stage_id.closed", "=", "True"),("closed_date", "<=", date_archive)])
+        
+        for ticket in closed:
+            ticket.active = False
```

Comment:
Suggested improvement:
```suggestion
        if not interval_number:
            return
        closed = self.search([
            ("stage_id.closed", "=", "True"), 
            ("closed_date", "<=", date_archive)
        ])
        if closed:
            closed.write({"active": False})
```